test/pylib/rest_client.py

```diff
@@ -0,0 +1,80 @@
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+"""Asynchronous helper for Scylla REST API operations.
+"""
+from abc import ABCMeta, abstractmethod
+import os.path
+from typing import Optional
+import aiohttp
+
+
+class RESTClientBase(metaclass=ABCMeta):
+    """Base class for REST API operations"""
+
+    def __init__(self, connector: aiohttp.BaseConnector = None):
+        self.session: aiohttp.ClientSession = aiohttp.ClientSession(connector = connector)
+
+    async def close(self) -> None:
+        """End session"""
+        await self.session.close()
+
+    @abstractmethod
+    def _resource_uri(self, resource: str, host: Optional[str] = None) -> str:
+        return ""
+
+    async def get(self, resource: str, host: Optional[str] = None) -> aiohttp.ClientResponse:
```

Comment:
If you have an `Optional` parameter in a base class only so it can be always passed as `None` to one subclass and as not `None` to another, there's a problem with your class hierarchy.

Look: `TCPRESTClient` has this: `def _resource_uri(self, resource: str, host: Optional[str] = None)` but it's always an error to pass `host = None` for this subclass. I recommend reading about the Liskov Substitution Principle, which this hierarchy blatantly breaks.

You need to structure this differently. If you want to use inheritance the hierarchy should preserve LSP. But in this case you just want to achieve some code reuse which can be done with simple composition like extracting logic to helper functions.